core/dbt/task/run_operation.py

```diff
@@ -49,6 +49,9 @@ def _run_unsafe(self, package_name, macro_name) -> "agate.Table":
                 macro_name, project=package_name, kwargs=macro_kwargs, macro_resolver=self.manifest
             )
 
+        if isinstance(res, str):
+            return None
```

Comment:
Note: When there is no explicit `return` function call in the macro — like for the `select_something` test — `res` is a `str` of newline characters and spaces, at least as observed with the Postgres and SQLite adapters.

Reply:
A more robust check is `not isinstance(res, agate.Table)` but there appears to be deliberate avoidance of `import agate`. Let me know if that is preferred.